types/vega/vegapulseAir/uplink.js

```diff
@@ -0,0 +1,319 @@
+      const { customFields } = device;
+      let scaleLength = null;
+      let sensorDistance = 0;
+
+      if (customFields.containerHeight !== undefined) {
+        scaleLength = Number(device.customFields.containerHeight);
+      }
+
+      if (customFields.installationOffset !== undefined) {
+        sensorDistance = Number(device.customFields.installationOffset);
+      }
+
+      if (scaleLength !== null) {
+        const percentExact =
+          (100 / scaleLength) * (scaleLength - (distance - sensorDistance));
+        let fillLevel = Math.round(percentExact);
+        if (fillLevel > 100) {
+          fillLevel = 100;
+        } else if (fillLevel < 0) {
+          fillLevel = 0;
+        }
+        return fillLevel;
+      }
+    }
+  }
+  return undefined;
+}
+
+function consume(event) {
+  const bytes = Hex.hexToBytes(event.data.payloadHex);
+  const distance = {};
+  const temperature = {};
+  const lifecycle = {};
+  const gps = {};
+  const system = {};
+  const schedule = {};
+
+  const packetIdentifier = bytes[0];
+
+  if (packetIdentifier === 2 || packetIdentifier === 3 || packetIdentifier === 4 || packetIdentifier === 5) {
+    lifecycle.namurState = namurStatus(bytes[1]);
+    distance.distance = bytesToFloat(bytes.slice(2, 6));
+    distance.unit = bytes[6];
+    lifecycle.batteryLevel = bytes[7];
+    temperature.temperature = bytesToInt16(bytes.slice(8, 10)) / 10;
+
+    if (packetIdentifier === 2) {
+      distance.inclinationDegree = bytes[10];
+    } else if (packetIdentifier === 3) {
+      gps.latitude = bytesToFloat(bytes.slice(10, 14));
+      gps.longitude = bytesToFloat(bytes.slice(14, 18));
+      distance.inclinationDegree = bytes[18];
+    } else if (packetIdentifier === 4) {
+      distance.detailedState = bytesToUInt32(bytes.slice(10, 14));
+      distance.inclinationDegree = bytes[14];
+    } else if (packetIdentifier === 5) {
+      gps.latitude = bytesToFloat(bytes.slice(10, 14));
+      gps.longitude = bytesToFloat(bytes.slice(14, 18));
+      distance.detailedState = bytesToUInt32(bytes.slice(18, 22));
+      distance.inclinationDegree = bytes[22];
+    }
+  } else if (packetIdentifier === 6) {
+    lifecycle.namurState = namurStatus(bytes[1]);
+    gps.latitude = bytesToFloat(bytes.slice(2, 6));
+    gps.longitude = bytesToFloat(bytes.slice(6, 10));
+  } else if (packetIdentifier === 7) {
+    lifecycle.namurState = namurStatus(bytes[1]);
+    distance.detailedState = bytesToUInt32(bytes.slice(2, 6));
+  } else if ((packetIdentifier >= 8 && packetIdentifier <= 15) || (packetIdentifier >= 18 && packetIdentifier <= 22)) {
+    let position = 1; // skip identifier
+    if ([10, 11, 14, 15, 19, 20, 21, 22].indexOf(packetIdentifier) !== -1) {
+      lifecycle.namurState = namurStatus(bytes[position]);
+      position++;
+    } else {
+      lifecycle.namurState = namurStatus(0);
+    }
+
+    if ([8, 9, 10, 11, 12, 13, 14, 15, 18, 19].indexOf(packetIdentifier) !== -1) {
```

Comment:
is there some meaning behind this array? if yes assign it to a variable to describe it